conftest.py

```diff
@@ -73,3 +75,77 @@ def undo(self):
 @pytest.fixture(autouse=True)
 def default_patches(request):
     return DefaultPatches(request)
+
+
+@pytest.fixture()
+def set_env_variables():
+    os.environ["BORG_CHECK_I_KNOW_WHAT_I_AM_DOING"] = "YES"
+    os.environ["BORG_DELETE_I_KNOW_WHAT_I_AM_DOING"] = "YES"
+    os.environ["BORG_PASSPHRASE"] = "waytooeasyonlyfortests"
+    os.environ["BORG_SELFTEST"] = "disabled"
+
+
+class ArchiverSetup:
+    EXE: str = None  # python source based
+    FORK_DEFAULT = False
+    prefix = ""
+    BORG_EXES = []
+
+    def __init__(self):
+        self.archiver = None
+        self.tmpdir = str
+        self.repository_path = str
+        self.repository_location = str
+        self.input_path = str
+        self.output_path = str
+        self.keys_path = str
+        self.cache_path = str
+        self.exclude_file_path = str
+        self.patterns_file_path = str
+        self.old_wd = str
+
+
+@pytest.fixture()
+def archiver(tmp_path, set_env_variables):
+    archiver = ArchiverSetup()
+    archiver.archiver = not archiver.FORK_DEFAULT and Archiver() or None
+    archiver.tmpdir = tmp_path
+    archiver.repository_path = os.fspath(tmp_path / "repository")
+    archiver.repository_location = archiver.prefix + archiver.repository_path
+    archiver.input_path = os.fspath(tmp_path / "input")
+    archiver.output_path = os.fspath(tmp_path / "output")
+    archiver.keys_path = os.fspath(tmp_path / "keys")
+    archiver.cache_path = os.fspath(tmp_path / "cache")
+    archiver.exclude_file_path = os.fspath(tmp_path / "excludes")
+    archiver.patterns_file_path = os.fspath(tmp_path / "patterns")
+    os.environ["BORG_KEYS_DIR"] = archiver.keys_path
+    os.environ["BORG_CACHE_DIR"] = archiver.cache_path
+    os.mkdir(archiver.input_path)
+    os.chmod(archiver.input_path, 0o777)  # avoid troubles with fakeroot / FUSE
+    os.mkdir(archiver.output_path)
+    os.mkdir(archiver.keys_path)
+    os.mkdir(archiver.cache_path)
+    with open(archiver.exclude_file_path, "wb") as fd:
+        fd.write(b"input/file2\n# A comment line, then a blank line\n\n")
+    with open(archiver.patterns_file_path, "wb") as fd:
+        fd.write(b"+input/file_important\n- input/file*\n# A comment line, then a blank line\n\n")
+    archiver.old_wd = os.getcwd()
+    os.chdir(archiver.tmpdir)
+    yield archiver
+    os.chdir(archiver.old_wd)
+
+
+@pytest.fixture()
+def remote_archiver(archiver):
+    archiver.prefix = "ssh://__testsuite__"
+    archiver.repository_location = archiver.prefix + str(archiver.repository_path)
+    yield archiver
+
+
+@pytest.fixture()
+def binary_archiver(archiver):
+    if "binary" not in BORG_EXES:
+        pytest.skip("No borg.exe binary available")
+    archiver.EXE = "borg.exe"
+    archiver.FORK_DEFAULT = True
+    yield archiver
```

Comment:
I will do this and report back.

Reply:
ok, so how was the outcome?

Reply:
@ThomasWaldmann Sorry, I need a little more time with this one. Still learning about Vagrant, and building binaries. I took a look at the docs and into the scripts directory for more information, and was able to build the Borg.exe, however I am still working on getting to a place where I can test it. This will be my focus today.

Reply:
Update - I have a call with Manu tomorrow and we are going to discuss Vagrant and testing on a Borg binary.

Reply:
You don't need to do anything with vagrant. I just pointed you to the Vagrantfile, because you can see all commands there you need to build a binary for local testing.

Reply:
maybe you could even just use a downloaded 2.0.0beta binary and not build it yourself, see github releases page (assuming the current tests still fit the behaviour of the last beta).